x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -171,52 +195,63 @@ const DragDropInner = React.memo(function DragDropInner(
     draggable,
     dragging,
     setDragging,
+    setActiveDropTarget,
     isDragging,
     isNotDroppable,
     dragType = 'copy',
     dropType = 'add',
     dropTo,
     itemsInGroup,
+    dropTargetIdentifier,
+    activeDropTarget,
   } = props;
 
+  const activeDropTargetMatches =
+    activeDropTarget && isEqual(dropTargetIdentifier, activeDropTarget);
+
   const isMoveDragging = isDragging && dragType === 'move';
 
   const classes = classNames(
     'lnsDragDrop',
     {
       'lnsDragDrop-isDraggable': draggable,
       'lnsDragDrop-isDragging': isDragging,
-      'lnsDragDrop-isHidden': isMoveDragging,
+      'lnsDragDrop-isHidden': isMoveDragging && !activeDropTarget,
       'lnsDragDrop-isDroppable': !draggable,
-      'lnsDragDrop-isDropTarget': droppable && dragType !== 'reorder',
-      'lnsDragDrop-isActiveDropTarget': droppable && state.isActive && dragType !== 'reorder',
+      'lnsDragDrop-isDropTarget':
+        // (!activeDropTarget || activeDropTargetMatches) makes sure only the currently active drop target is highlighted, not all of them
+        // because this part is not implemented yet
+        droppable && (!activeDropTarget || activeDropTargetMatches) && dragType !== 'reorder',
+      'lnsDragDrop-isActiveDropTarget':
+        droppable && (state.isActive || activeDropTargetMatches) && dragType !== 'reorder',
       'lnsDragDrop-isNotDroppable': !isMoveDragging && isNotDroppable,
       'lnsDragDrop-isReplacing': droppable && state.isActive && dropType === 'replace',
     },
     state.dragEnterClassNames
   );
 
-  const dragStart = (e: DroppableEvent) => {
+  const dragStart = (e?: DroppableEvent) => {
```

Comment:
I think it's a one-liner to start the drag from keyboard, and might improve readability to have a separate "start keyboard drag" function. `setDragging(value);`